### DISCUSSION TITLE
Recommended way to generate introspection file

### DISCUSSION THREAD

**Discussion by beeirl** [urql-graphql/urql#3183] [Q&A] [answered]
I am currently working on the schema awareness of my URQL `client` and was able to get everything working as described in the [docs](https://formidable.com/open-source/urql/docs/graphcache/schema-awareness/). I am now wondering what's the best way to generate the introspection file? My react-based web application currently fetches the gql schema everytime a user opens my web app. Would you recommend to generate the file at build time? I wasn't really able to find best practices in the docs nor here in the repo.

**Comment by JoviDeCroock** [comment, marked as answer]
The script mentioned [here](https://formidable.com/open-source/urql/docs/graphcache/schema-awareness/#getting-your-schema) is meant as something you would do at build time, not something you do during the lifetime of your application i.e. you can do `node ./script.mjs && build || dev`

**Reply by beeirl** [reply]
Gotcha. I ended up using `graphql-codegen` to derive it directly from my schema at build time. Works like a charm. Thanks for getting back to me!
